doc/index.rst

```diff
@@ -0,0 +1,599 @@
+   See :py:class:`typing.ContextManager`. In ``typing`` since 3.5.4.
+
+.. class:: Coroutine
+
+   See :py:class:`typing.Coroutine`. In ``typing`` since 3.5.3.
+
+Protocols
+~~~~~~~~~
+
+.. class:: SupportsAbs
+
+   See :py:class:`typing.SupportsAbs`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol in Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+.. class:: SupportsBytes
+
+   See :py:class:`typing.SupportsBytes`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol in Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+.. class:: SupportsComplex
+
+   See :py:class:`typing.SupportsComplex`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol in Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+.. class:: SupportsFloat
+
+   See :py:class:`typing.SupportsFloat`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol in Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+.. class:: SupportsIndex
+
+   See :py:class:`typing.SupportsIndex`. In ``typing`` since 3.8.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol in Python 3.11 and lower.
+
+   .. versionchanged:: 4.6.0
+
+      Backported the performance improvements from Python 3.12.
+
+.. class:: SupportsInt
+
+   See :py:class:`typing.SupportsInt`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol in Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+.. class:: SupportsRound
+
+   See :py:class:`typing.SupportsRound`.
+
+   ``typing_extensions`` backports a more performant version of this
+   protocol in Python 3.11 and lower.
+
+   .. versionadded:: 4.6.0
+
+Decorators
+~~~~~~~~~~
+
+.. decorator:: dataclass_transform(*, eq_default=False, order_default=False,
+                                   kw_only_default=False, frozen_default=False,
+                                   field_specifiers=(), **kwargs)
+
+   See :py:func:`typing.dataclass_transform`. In ``typing`` since 3.11.
+
+   Python 3.12 adds the ``frozen_default`` parameter; ``typing_extensions``
+   backports this parameter.
+
+   .. versionadded:: 4.1.0
+
+   .. versionchanged:: 4.2.0
+
+      The ``field_descriptors`` parameter was renamed to ``field_specifiers``.
+      For compatibility, the decorator now accepts arbitrary keyword arguments.
+
+   .. versionchanged:: 4.5.0
+
+      The ``frozen_default`` parameter was added.
+
+.. decorator:: deprecated(msg, *, category=DeprecationWarning, stacklevel=1)
+
+   See :pep:`702`.
+
+   .. versionadded:: 4.5.0
+
+.. decorator:: final
+
+   See :py:func:`typing.final`. In ``typing`` since 3.8.
+
+   Since Python 3.11, this decorator supports runtime introspection
+   through the ``__final__`` attribute; ``typing_extensions.final``
+   backports this feature.
+
+   .. versionchanged:: 4.1.0
+
+      The ``__final__`` attribute is now set on decorated objects.
```

Comment:
Maybe something like:

> The decorator now attempts to set the `__final__` attribute on decorated objects. AttributeErrors and TypeErrors are caught when trying to set this attribute.

Reply:
```suggestion
      The decorator now attempts to set the ``__final__`` attribute on decorated objects. 
```
Left out the second sentence for brevity; details of behavior can be in the CPython docs.